### PR TITLE
added bounding capabilities for plot and plot_eps

### PR DESCRIPTION
**Pull request by alec-flexcompute**

### CODE REVIEW

File: tidy3d/components/simulation.py [hunk 1]
```diff
@@ -2217,6 +2359,65 @@ def _get_structures_plane(
...
+            Vertical bounds for plotting
+
+        Returns
+        -------
+        List[Tuple[:class:`.AbstractMedium`, shapely.geometry.base.BaseGeometry]]
+            List of shapes and mediums on the plane.
+        """
+        (h_min, h_max), (v_min, v_max) = hlim, vlim
+        h_length, v_length = h_max - h_min, v_max - v_min
+        if z == None:
```
Comment (tylerflex) [inline tidy3d/components/simulation.py:2395]: there exists a `self.unpop_axis()` [docs](https://docs.flexcompute.com/projects/tidy3d/en/latest/_autosummary/tidy3d.Sphere.html?highlight=unpop%20axis#tidy3d.Sphere.unpop_axis) that could be useful here to simplify things. Here's one suggestion

```py
(h_min, h_max), (v_min, v_max) = hlim, vlim

# get center and size in the h, v "basis"
h_center = (h_min + h_max) / 2.0
v_center = (v_min + v_max) / 2.0
h_size = h_max + h_min 
v_size = v_max + v_min

axis, center_normal = self.parse_xyz_kwargs(x=x, y=y, z=z)
center = self.unpop_axis(center_normal, (h_center, v_center), axis=axis)
size = self.unpop_axis(0.0, (h_size, v_size), axis=axis)

# continue with line 2410
```

unpop axis is the inverse of pop axis and I think handles conversion from h, v to x, y, z more elegantly.

**Review by tylerflex — CHANGES_REQUESTED** [review]
Thanks @alec-flexcompute , I had a few refactors and minor comments. Feel free to take your time and work on this and then reassign me to review. Otherwise looks good and it's almost ready!

**Comment by tylerflex** [comment]
Also, to check some of the formatting tests locally, make sure to run 
```
python lint.py
```
the tests are failing now due to some warnings related to the `None` checking I mention.

File: tidy3d/components/simulation.py [hunk 2]
```diff
@@ -1330,20 +1332,46 @@ def plot(
             Opacity of the monitors. If ``None``, uses Tidy3d default.
         ax : matplotlib.axes._subplots.Axes = None
             Matplotlib axes to plot on, if not specified, one is created.
+        hlim : Tuple[float, float] = None
+            Horizontal bounds for plotting (x range if z specified, y range if x specified, etc)
```
[the remark this answers is not in the document]
Comment (alec-flexcompute) [reply tidy3d/components/simulation.py:1336]: I'm thinking of writing xy or xz planes instead, unless it's understood what a y-plane is

[the remark this answers is not in the document]
Comment (tylerflex) [reply tidy3d/components/simulation.py:1336]: I think y plane -> xz plane is probably more clear yea

File: tidy3d/components/simulation.py [hunk 3]
```diff
@@ -2180,9 +2316,15 @@ def _set_plot_bounds(self, ax: Ax, x: float = None, y: float = None, z: float =
         _, (xmin, ymin) = self.pop_axis(self.bounds_pml[0], axis=axis)
         _, (xmax, ymax) = self.pop_axis(self.bounds_pml[1], axis=axis)
 
-        if xmin != xmax:
+        if hlim != None:
```
[the remark this answers is not in the document]
Comment (alec-flexcompute) [reply tidy3d/components/simulation.py:2319]: Yes this works and is much simpler. Fixed!

File: tidy3d/components/simulation.py [hunk 4]
```diff
@@ -1388,20 +1418,50 @@ def plot_eps(  # pylint:disable=too-many-arguments
...
-        ax = self.plot_symmetries(ax=ax, x=x, y=y, z=z)
-        ax = self.plot_pml(ax=ax, x=x, y=y, z=z)
-        ax = self._set_plot_bounds(ax=ax, x=x, y=y, z=z)
-        ax = self.plot_boundaries(ax=ax, x=x, y=y, z=z)
+        # if no hlim and/or vlim given, the bounds will then be the usual pml bounds
+        axis, _ = self.parse_xyz_kwargs(x=x, y=y, z=z)
+        _, (hmin, vmin) = self.pop_axis(self.bounds_pml[0], axis=axis)
+        _, (hmax, vmax) = self.pop_axis(self.bounds_pml[1], axis=axis)
+
+        new_h_lim, new_v_lim = hlim, vlim
```
[the remark this answers is not in the document]
Comment (alec-flexcompute) [reply tidy3d/components/simulation.py:1426]: Yep! Much simpler and changed.

File: tidy3d/components/simulation.py (hunk 1 above)
Comment (alec-flexcompute) [reply tidy3d/components/simulation.py:2395]: h_size = h_max - h_min but yes otherwise this works elegantly. changed.

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 
 ### Added
+- Added hlim and vlim kwargs to `Simulation.plot()` and `Simulation.plot_eps()` for setting horizontal and veritcal plot limits.
```
Comment (tylerflex) [inline CHANGELOG.md:9]: could you put the same single backslash quotes around `hlim` and `vlim`? Thanks

File: tidy3d/components/simulation.py [hunk 6]
```diff
@@ -1330,19 +1332,38 @@ def plot(
             Opacity of the monitors. If ``None``, uses Tidy3d default.
         ax : matplotlib.axes._subplots.Axes = None
             Matplotlib axes to plot on, if not specified, one is created.
+        hlim : Tuple[float, float] = None
+            The x range if plotting on y or z planes, y range if plotting on x plane.
```
Comment (tylerflex) [inline tidy3d/components/simulation.py:1336]: do we want to change to `xy` or `xz` planes, for example? probably clearer like you said.

File: tidy3d/components/simulation.py [hunk 7]
```diff
@@ -1330,19 +1332,38 @@ def plot(
...
+        _, (hmax, vmax) = self.pop_axis(self.bounds_pml[1], axis=axis)
+
+        # account for unordered limits
+        if hlim is None:
+            hlim = (hmin, hmax)
+        if vlim is None:
+            vlim = (vmin, vmax)
+
+        if hlim[0] > hlim[1]:
+            raise Tidy3dError("Error: hmin > hmax")
```
Comment (tylerflex) [inline tidy3d/components/simulation.py:1357]: in error messages (things that get printed more generally) we put regular single quotes `'` around variables, so 
```
Error: 'hmin' > 'hmax'
```
There's a discussion [here](https://www.notion.so/flexcompute/Python-Style-Guide-afac5dbbf949463e89c18fe39b7453c9) if you want to read about it.

File: tidy3d/components/simulation.py [hunk 8]
```diff
@@ -1415,21 +1465,26 @@ def plot_structures(
...
+            The z range if plotting on x or y planes, y plane if plotting on z plane.
 
         Returns
         -------
         matplotlib.axes._subplots.Axes
             The supplied or created matplotlib axes.
         """
 
-        medium_shapes = self._get_structures_plane(structures=self.structures, x=x, y=y, z=z)
+        medium_shapes = []
```
Comment (tylerflex) [inline tidy3d/components/simulation.py:1479]: probably we can remove this line?

File: tidy3d/components/simulation.py [hunk 9]
```diff
@@ -2217,6 +2337,68 @@ def _get_structures_plane(
...
+        structures : List[:class:`.Structure`]
+            list of structures to filter on the plane.
+        x : float = None
+            position of plane in x direction, only one of x, y, z must be specified to define plane.
+        y : float = None
+            position of plane in y direction, only one of x, y, z must be specified to define plane.
+        z : float = None
+            position of plane in z direction, only one of x, y, z must be specified to define plane.
+        hlim : Tuple[float,float] = None
+            Horizontal bounds for finding structures
```
Comment (tylerflex) [inline tidy3d/components/simulation.py:2362]: probably we can use the same discussion from the `.plot()` docstring here about which dimensions these refer to given what plane we are looking at.

File: tidy3d/plugins/adjoint/components/simulation.py [hunk 10]
```diff
@@ -317,6 +319,8 @@ def plot_eps(
         alpha: float = None,
         source_alpha: float = None,
         monitor_alpha: float = None,
+        hlim: Tuple[float, float] = None,
+        vlim: Tuple[float, float] = None,
```
Comment (tylerflex) [inline tidy3d/plugins/adjoint/components/simulation.py:323]: we should also pass these into the `sim.plot()` and `sim.plot_eps()` calls in these methods. basically, this is making simulation plotting work in the `JaxSimulation` subclass of `Simulation`.

**Review by tylerflex — CHANGES_REQUESTED** [review]
Thanks @alec-flexcompute , just a few other minor things.

**Review by tylerflex — APPROVED** [review]
Thanks @alec-flexcompute ! Merging this

**Pull request merged**